examples/machine_learning/neural_network_train.py

```diff
@@ -0,0 +1,308 @@
+# 
+# ## Neural Networks
+# 
+# In the original paper, a linear regression is found using the above descriptors with a 10-fold cross validated Root Mean Squared Error (RMSE) of 0.179 eV:
+# 
+# $E_{ads} \approx a*CE_{local} + b*\mu_{adsorbate} + c*MADs + d$
+# 
+# In this notebook, we replicate the results of this linear regression using a neural network. We proceed through a -step process:
+# 
+# 1. We import the adsorption data along with the above descriptors descriptor (available in the supporting information of [the paper](https://advances.sciencemag.org/content/5/9/eaax5101.abstract))
+# 2. We create a function to programatically create a neural network using TensorFlow
+# 3. We optimze several neural network hyperparameters using Scikit-Learn's cross-validated grid search implementation
+# 4. We train a final neural network using the tuned hyperparameters, and evaluate it by:
+#     a. Learning curves for the train / test set
+#     b. A parity plot of predicted versus ground-truth adsorption energy
+
+# # Setting Up
+
+# In[]:
+
+
+import random
+import logging
+import os
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import sklearn.model_selection
+import subprocess
+import tensorflow as tf
+import IPython
+from tensorflow.keras.layers import Dense, BatchNormalization, InputLayer
+from tensorflow.keras.wrappers.scikit_learn import KerasRegressor
+
+
+# In[]:
+
+
+# Check if GraphViz is installed; used to plot the NN architecture later on.
+rc = subprocess.call(['dot', '-v'])
+if rc == 0:
+    dot_exists = True
+else:
+    dot_exists = False
+
+
+# In[]:
+
+
+# Random seeds for reproducibility
+np.random.seed(1234)
+random.seed(1234)
+
+
+# Next, we'll get everything into a Pandas dataframe.
+
+# In[]:
+
+
+# Load data into a Pandas dataframe for convenience
+df = pd.read_csv("Data.csv", sep=",")
+y_cols = ["PBE_BE_eV"]
+x_cols = ["CE_Local_eV", "ChemPot_eV", "MADS_eV"]
+
+train_x = df[x_cols]
+train_y = df[y_cols]
+
+
+# ## Hyperparameter Optimization
+# 
+# In this example, we use Scaled-Exponential Linear Unit activation functions, which [provably](https://arxiv.org/pdf/1706.02515.pdf) result in a self-normalizing effect as information flows through a neural network's layers, provided the network is sufficiently deep. One assumption made in the proof is the use of LeCunn's initialization strategy for weights and biases.
+
+# In[]:
+
+
+# Weight initialization strategy
+initializer = tf.keras.initializers.lecun_normal() 
```

Comment:
Per our discussion today - some of the details re choices for the initialization and/or activation deserve more explanation. Pls add.

Reply:
Dropped the selu / lecun stuff. We're now just running a basic relu neural network. Most people are more familiar with this style of network than they are with selu-based nets